ui.pyw

```diff
@@ -245,12 +252,124 @@ class DbTab(Tab):
             tkMessageBox.showerror("Adding user to restaurants failed", "Either we failed to connect to the database, or the user or one of the restaurants doesn't exist or the mapping already exists")
             return
 
+class UserDialog(Toplevel, object):
+    __metaclass__ = ABCMeta
+    @abstractmethod
+    def button_ok_action(self): pass
+    @abstractmethod
+    def fill_initial_values(self): pass
+
+    def __init__(self, parent, db):
+        Toplevel.__init__(self, parent)
+        self.parent = parent
+        self._db = db
+        self.transient(parent)
+
+        # Setting font since the default is weird and jumbles-up hebrew with apostrophe (stackoverflow.com/q/34220597/2134702)
+        self.restaurants_list = Listbox(self, selectmode='multiple', exportselection=0, font=('Tahoma', 8))
+        self.restaurants_list_scrollbar = Scrollbar(self, command=self.restaurants_list.yview)
+        self.restaurants_list['yscrollcommand'] = self.restaurants_list_scrollbar.set
+        self.choose_restaurants_label = Label(self, text="Restaurants:")
+
+        self.user_label = Label(self, text="Person Name:")
+        self.user_text = Text(self, height=1, width=18)
+
+        self.ok_button = Button(self, text="OK", command=self._button_ok)
+        self.cancel_button = Button(self, text="Cancel", command=self._button_cancel)
+
+        self.columnconfigure(0, weight=0)
+        self.columnconfigure(1, weight=0)
+        self.columnconfigure(2, weight=0)
+        self.columnconfigure(3, weight=0)
+        self.columnconfigure(4, weight=1)
+        self.rowconfigure(0, weight=0)
+        self.rowconfigure(1, weight=0)
+        self.rowconfigure(2, weight=1)
+
+        #layout
+        self.user_label.grid(row=0, column=0, sticky=E + N)
+        self.user_text.grid(row=0, column=1, sticky=W + N)
+        self.choose_restaurants_label.grid(row=2, column=0, sticky=E + N)
+        self.restaurants_list.grid(row=2, column=1, sticky=W + E + S + N)
+        self.restaurants_list_scrollbar.grid(row=2, column=2, sticky=W + S + N)
+        self.ok_button.grid(row=3, column=1, sticky=E + S + N)
+        self.cancel_button.grid(row=3, column=2, sticky=W + S + N)
+
+        #fill up fields
+        available_restaurants = set(self._db.get_all_restaurants())
+        #chosen_restaurants = set(CrossReferencer.find_restaurants({user}))
+        self.restaurants_list.delete(0, END)
+        for item in available_restaurants:
+            self.restaurants_list.insert(END, item)
+
+        self.fill_initial_values()
+
+        self.wait_visibility()
+        self.grab_set() #makes window modal
+
+    def _button_ok(self):
```

Comment:
Yoda speak is vacation not in cities kind of people do. (isn't _ok_button better?)